src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java

```diff
@@ -100,13 +100,59 @@ public DefaultSerializer(final ILogger logger) {
             }
             
             jsonBackedObject.additionalDataManager().setAdditionalData(rawObject);
+            setChildAdditionalData(jsonBackedObject,rawObject);
             return jo;
         } else {
             logger.logDebug("Deserializing a non-IJsonBackedObject type " + clazz.getSimpleName());
             return jsonObject;
         }
     }
 
+    /**
+     * Recursively sets additional data for each child object
+     *
+     * @param serializedObject   the parent object whose children will be iterated to set additional data
+     * @param rawJson            the raw json
+     */
+    @SuppressWarnings("unchecked")
+    private void setChildAdditionalData(IJsonBackedObject serializedObject, JsonObject rawJson) {
```

Comment:
No unit tests for this method. Can they be added?

Reply:
Thanks for bringing it up. I will add the Unit test.

Reply:
I have raised a separate PR for this - #114 .